src/meta/processors/admin/LeaderBalanceProcessor.cpp

```diff
@@ -6,17 +6,437 @@
 
 
 #include "meta/processors/admin/LeaderBalanceProcessor.h"
-#include "meta/processors/admin/Balancer.h"
+
+DEFINE_double(leader_balance_deviation, 0.05, "after leader balance, leader count should in range "
+                                              "[avg * (1 - deviation), avg * (1 + deviation)]");
 
 namespace nebula {
 namespace meta {
 
-void LeaderBalanceProcessor::process(const cpp2::LeaderBalanceReq&) {
-    auto ret = Balancer::instance(kvstore_)->leaderBalance();
+void LeaderBalanceProcessor::process(const cpp2::LeaderBalanceReq& req) {
+    space_ = req.get_space_id();
+    LOG(INFO) << "Space " << req.get_space_id();
+    auto ret = leaderBalance();
     handleErrorCode(ret);
     onFinished();
 }
 
+cpp2::ErrorCode LeaderBalanceProcessor::leaderBalance() {
+    folly::Promise<Status> promise;
+    auto future = promise.getFuture();
+    auto spaceInfoRet = MetaCommon::getSpaceInfo(kvstore_, space_);
+    if (!spaceInfoRet.ok()) {
```

Comment:
We only support balance leader for one space? Emm, how to balance leader for all space?

Reply:
yep currently we only support balance leader one by one

Reply:
We'd better keep it. For example if `balance leader space` then we only balance for that space, if `balance leader` then all spaces.